### PR TITLE
Check resource name

### PR DESCRIPTION
**Pull request by legionus**
Fix for RHBZ: https://bugzilla.redhat.com/show_bug.cgi?id=1528613

### CODE REVIEW

File: pkg/dockerregistry/server/token.go [hunk 1]
```diff
@@ -30,6 +31,22 @@ const anonymousToken = "anonymous"
...
+	params := req.URL.Query()
+	if len(params.Get("scope")) > 0 {
+		accessRecords := auth.ResolveScopeSpecifiers(ctx, params["scope"])
+		for _, access := range accessRecords {
+			switch access.Resource.Type {
+			case "repository", "signature":
+				_, _, err := getNamespaceName(access.Resource.Name)
+				if err != nil {
+					context.GetRequestLogger(ctx).Debugf("auth token request for unsupported resource name: %s", access.Resource.Name)
+					t.writeError(w, req)
```
Comment (dmage) [inline pkg/dockerregistry/server/token.go]: Don't we want to send an error_description?

File: pkg/dockerregistry/server/util.go [hunk 2]
```diff
@@ -21,9 +22,12 @@ import (
...
 
 func getNamespaceName(resourceName string) (string, string, error) {
-	repoParts := strings.SplitN(resourceName, "/", 2)
+	repoParts := strings.Split(resourceName, "/")
 	if len(repoParts) != 2 {
-		return "", "", ErrNamespaceRequired
+		return "", "", distribution.ErrRepositoryNameInvalid{
+			Name:   resourceName,
+			Reason: fmt.Errorf("invalid repository name %q: it must be of the format <project>/<name>", resourceName),
+		}
```
Comment (dmage) [inline pkg/dockerregistry/server/util.go]: The resulting error will be weird: `repository name %q invalid: invalid repository name %q: it must be of the format <project>/<name>`

**Comment by bparees** [comment]
@legionus is this related to the BZ about pushes to multisegment paths?  Can you please link the bug in your commit and PR description?  If it's not that, some other description of what this is addressing would be helpful.

**Comment by legionus** [comment]
@bparees Yes, I updated the description and sent a link on this PR to bugzilla.

**Pull request merged**